source/Halibut/Transport/Protocol/ControlMessageReader.cs

```diff
@@ -59,7 +59,7 @@ internal async Task<string> ReadUntilNonEmptyControlMessageAsync(Stream stream)
 
         internal async Task<string> ReadControlMessageAsync(Stream stream)
         {
-            var cts = GetCancellationTokenSourceFromStreamReadTimeout(stream);
+            using var cts = GetCancellationTokenSourceFromStreamReadTimeout(stream);
```

Comment:
In previous versions of VS, you could setup code analysis to catch instances where disposable objects were not being disposed.

Maybe this is something we should think about doing in this solution some time in the future?

Reply:
Oops this was me

Reply:
https://github.com/dotnet/runtime/issues/29970 This person talks about disposing these timer CTS as being important.